test/e2e/issues_test.go

```diff
@@ -339,6 +339,65 @@ spec:
 		MustApplyYAML(eetestUpdated)
 		FieldShouldContainWithTimeout("eetests.e2e.hnc.x-k8s.io", nsChild, "eetest-sample", ".spec.foo", "bar", 30)
 	})
+
+	It("Should allow objects to be propagated only when a selector is set in AllowPropagate mode - issue #16", func() {
+		// Create a simple structure and get an object propagated
+		CreateNamespace(nsParent)
+		CreateNamespace(nsChild)
+		MustRun("kubectl hns set", nsChild, "--parent", nsParent)
+
+		MustRun("kubectl -n", nsParent, "create secret generic my-creds --from-literal=password=iamteamb")
+		// secret does not show up in child namespae because we haven’t configured HNC to propagate secrets in HNCConfiguration.
+		time.Sleep(2 * time.Second)
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+		// after setting Propagate mode, the object shows up in the child namespace
+		MustRun("kubectl hns config set-resource secrets --mode Propagate --force")
+		time.Sleep(2 * time.Second)
```

Comment:
Not needed, `RunShouldContain` repeats for 2s. All remaining `Sleeps` can and should be removed as long as the preceding command changes the state (e.g. from propagated to unpropagated or vice versa), which I believe is true. You only need to sleep if you're confirming that _nothing_ has changed, which we generally avoid doing.

Reply:
Understood, thanks